### DISCUSSION TITLE
Packaging status

### DISCUSSION THREAD

**Discussion by tim77** [yshui/picom#228] [General]
Sorry if this duplicate but i didn't found global tracking packaging status or something like that. Just FYI: compton (new version) packaged for Fedora, but under new package name `compton-ng`.

- [Package start page](https://apps.fedoraproject.org/packages/compton-ng) (buggy, not showing at this time)
- [Source](https://src.fedoraproject.org/rpms/compton-ng)

How to install:

```shell
sudo dnf install compton-ng
```

**Comment by yshui** [comment]
Alright, let's turn this into a packaging status ticket.

From what I know, this `compton` is also officially packaged in:
* [ArchLinux](https://www.archlinux.org/packages/community/x86_64/compton/)
* [VoidLinux](https://github.com/void-linux/void-packages/blob/master/srcpkgs/compton/template)
* [NixOS](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/window-managers/compton/default.nix)
* [NetBSD](https://ftp.netbsd.org/pub/pkgsrc/current/pkgsrc/x11/compton/README.html)

Don't know if I am missing any.

**Reply by rbtylee** [reply]
I package it for [Bodhi Linux](https://www.bodhilinux.com/).

- [5.x](http://packages.bodhilinux.com/bodhi/pool/b5main/p/picom/)
- [6.x](http://packages.bodhilinux.com/bodhi/pool/b6main/p/picom/) _64 bit only_

**Comment by jenav** [comment]
Well, I'm keeping personal builds of stable and git releases for openSUSE 15.0, 15.1 and Tumbleweed on OBS:

- [compton](https://build.opensuse.org/package/show/home:jenav/compton)
- [picom-git](https://build.opensuse.org/package/show/home:jenav/picom-git)

Edit: update urls after package rename

**Comment by CamilleScholtz** [comment]
I'm keeping personal builds (based on git master) for CRUX: https://github.com/onodera-punpun/crux-ports/tree/master/compton

**Comment by ncmprhnsbl** [comment]
handy resource:
https://repology.org/projects/?search=compton
https://repology.org/project/compton/versions
https://repology.org/project/compton/packages

**Comment by yshui** [comment]
@ncmprhnsbl that's very helpful, thanks

**Comment by Vladimir-csp** [comment]
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=908533

**Comment by avindra** [comment]
This `-ng` pattern propagates itself in software causing unnecessary fragmentation. to truly represent the development of compton, we would have to be on `compton-ng-5` or something like that by now.

The package maintainers adding `-ng` are just following an odd convention that should be abandoned as it creates confusion amongst users and packagers alike.

**Comment by avindra** [comment]
For the record, I'm looking into replacing the `compton` package in openSUSE with the builds that @jenav has been maintaining :  https://build.opensuse.org/request/show/729602

(i.e., no `-ng` suffix).

**Comment by yshui** [comment]
@avindra I am aware, that's why there is #222

**Comment by avindra** [comment]
Here you can see the spread of `compton` across distros:

https://repology.org/project/compton/versions

It seems that fedora is the only distribution so far that is using `compton-ng` thankfully.

https://repology.org/project/compton-ng/versions

If a new name comes out of #222 then we can all happily agree on that.

**Comment by tim77** [comment]
> Here you can see the spread of compton across distros:

> It seems that fedora is the only distribution so far that is using compton-ng thankfully.

What a funny this guy @avindra :laughing: Indeed, because most of distros like Debian, Ubuntu and your openSUSE is still using old 0.1.0 version. What you trying to prove by posting this BS? You think someone will respecting you if you will continue posting you trash posts, thumbs down on every my post and hating Fedora and Fedora maintainers overall? Try spend your time to something useful, like packaging new compton for openSUSE. Oh wait, you already copypasted my spec file. :laughing: GJ.

And why you worrying how it named in Fedora? This shouldn't worry you at all. This thread about packaging status, not renaming.

**Comment by avindra** [comment]
@tim77 _I_ had no involvement in the spec file. Regardless of who authored what, perhaps you should consider that both RPM based distros may _naturally_ converge on the same solution?

> hating Fedora and Fedora maintainers overall

this is a deluded view and perhaps a mental projection on your part. I have no qualms with Fedora, its users or community.

**Comment by tim77** [comment]
> perhaps you should consider that both RPM based distros may naturally converge on the same solution?

@avindra are you fine man? I should consult with you maybe before i push package in Fedora repos? And openSUSE **still** doesn't have this new forked copmton version in official repos.
Pushing this separate version of compton help Fedora users using new fork right now, in current Fedora release and not waiting for the next F31 because updating to major version is not allowed by updating policy. Maybe in is OK in openSUSE to update and broke user configs, but not for Fedora. And if upstream decide to not change name then this compton-ng turns into compton in future Fedora releases easily. No prob there.

Now please stop trash talking and flooding in **Packaging status** topic.

**Comment by yshui** [comment]
@tim77 I really appreciate the effort you took to package this compton for Fedora.

I am not fond of the `-ng` suffix, but that's my own fault to not take the initiative and rename this fork sooner.

There's really no need to say such things about each other.

**Comment by kgilmer** [comment]
I am not an official packager, however I have your variant of Compton here:

https://github.com/regolith-linux/speed-ricer
https://launchpad.net/~kgilmer/+archive/ubuntu/speed-ricer/

It is also shipped by default in Regolith Linux R1.2: https://regolith-linux.org

**Comment by jbeich** [comment]
Given #222 and #207 should downstream package `v7.4` (pre-rename) as `compton` update or (wait for `v8`) as new `picom` package? From FreeBSD POV if `compton` package is left as is there's no need to worry how breaking changes may affect consumers (e.g., compton-conf, lumina-session) or old graphics drivers.

**Comment by yshui** [comment]
@jbeich I don't think there is a need to retroactively rename older packages. Newer packages should ideally be renamed right away (package managers with functionalities like pacman's replace, should use that to update user to the new name), but it's also OK as long as they complete the rename before v10

**Comment by jbeich** [comment]
> ... update user to the new name)

The question is whether to inherit users from chjj's compton or not. Before this fork was renamed some distributions have already switched existing package to yshui's compton (the new upstream). However, those that didn't can maintain both picom **and** compton. Once breaking changes land and things stabilize again (a la mplayer->mplayer2->mpv) the old package can be removed if no one wants to maintains it.

Currently, the latest release here uses the old name, complicating import as separate package from the start.

**Comment by yshui** [comment]
> those that didn't can maintain both picom and compton.

Yes, that's one of the points of this rename. However, those distros that have already made this switch probably won't be able to go back to chjj's compton. They probably have to wait a period of time after renaming to `picom`, before they can reintroduce the `compton` package.
